core/templates/pages/exploration-editor-page/services/user-email-preferences.service.ts

```diff
@@ -12,81 +12,100 @@
+      message_type: this.MESSAGE_TYPE_SUGGESTION,
+      mute: mute
+    }).then(()=> {
+      successCallback();
+    });
+  }
 
-angular.module('oppia').factory('UserEmailPreferencesService', [
-  '$http', 'AlertsService', 'ExplorationDataService',
-  'UrlInterpolationService',
-  function(
-      $http, AlertsService,
-      ExplorationDataService, UrlInterpolationService) {
-    var MESSAGE_TYPE_SUGGESTION = 'suggestion';
-    var MESSAGE_TYPE_FEEDBACK = 'feedback';
-    return {
-      init: function(
-          feedbackNotificationsMuted, suggestionNotificationsMuted) {
-        this.feedbackNotificationsMuted = feedbackNotificationsMuted;
-        this.suggestionNotificationsMuted = suggestionNotificationsMuted;
-      },
-      /**
-       * @return {boolean} Whether the feedback notification is muted.
-       */
-      areFeedbackNotificationsMuted: function() {
-        return this.feedbackNotificationsMuted;
-      },
-      /**
-       * @return {boolean} Whether the suggestion notification is muted.
-       */
-      areSuggestionNotificationsMuted: function() {
-        return this.suggestionNotificationsMuted;
-      },
-      /**
-       * Set the message type to feedback and mute to true or false.
-       * @param {boolean} mute - Whether the feedback notification is muted.
-       */
-      setFeedbackNotificationPreferences: function(mute) {
-        this.saveChangeToBackend({
-          message_type: MESSAGE_TYPE_FEEDBACK,
-          mute: mute
-        });
-      },
-      /**
-       * Set the message type to suggestion and mute to true or false.
-       * @param {boolean} mute - Whether the suggestion notification is muted.
-       */
-      setSuggestionNotificationPreferences: function(mute) {
-        this.saveChangeToBackend({
-          message_type: MESSAGE_TYPE_SUGGESTION,
-          mute: mute
-        });
-      },
-      /**
-       * Save the change of message_type and mute to backend.
-       * @param {object} requestParams - Info about message_type and mute.
-       */
-      saveChangeToBackend: function(requestParams) {
-        var that = this;
-        var emailPreferencesUrl = UrlInterpolationService.interpolateUrl(
-          '/createhandler/notificationpreferences/<exploration_id>', {
-            exploration_id: ExplorationDataService.explorationId
-          }
-        );
-        return $http.put(emailPreferencesUrl, requestParams).then(
-          function(response) {
-            var data = response.data;
-            AlertsService.clearWarnings();
-            that.init(
-              data.email_preferences.mute_feedback_notifications,
-              data.email_preferences.mute_suggestion_notifications);
-          }
-        );
-      }
-    };
+  /**
+   * Save the change of message_type and mute to backend.
+   * @param {RequestParam} requestParams - Info about message_type and mute.
+   */
+  saveChangeToBackend(requestParams: RequestParams) : Promise<void | object> {
```

Comment:
Promise<void | object> is not specific enough for what might be returned. Can you find the exact type?

Reply:
Done

Reply:
I found the exact type.
But lint tests fails when I use a type other an object.
Everything works as expected on the web app.
What should I do dI found the exact type.
But lint tests fails when I use a type other an object.
Everything works as expected on the web app.
What should I do disable lint checks for this or use the type object.
![Screenshot from 2021-02-25 12-59-05](https://user-images.githubusercontent.com/24855641/109118481-61bd4f00-7769-11eb-9f0a-d700e79c7ec0.png)
isable lint checks for this or use the type object.
![Screenshot from 2021-02-25 12-59-05](https://user-images.githubusercontent.com/24855641/109118481-61bd4f00-7769-11eb-9f0a-d700e79c7ec0.png)

Reply:
Done.